### PR TITLE
feat: add secretmanager sample for ConsumeEventNotification

### PR DESCRIPTION
**Pull request by Sita04**
## Description

Bug: b/281694482 
Doc update: cl/533479206

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [x] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [ ] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: secretmanager/src/main/java/secretmanager/ConsumeEventNotification.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    return log;
+  }
+
+  // Event payload. Mock of the actual Pub/Sub message.
+  public static class PubSubMessage {
+
+    byte[] data;
+    Map<String, String> attributes;
+    String messageId;
+    String publishTime;
```
Comment (anguillanneuf) [inline secretmanager/src/main/java/secretmanager/ConsumeEventNotification.java:50]: missing field `orderingKey`: https://cloud.google.com/pubsub/docs/reference/rest/v1/PubsubMessage

File: secretmanager/src/main/java/secretmanager/ConsumeEventNotification.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    public Map<String, String> getAttributes() {
+      return attributes;
+    }
+
+    public void setAttributes(Map<String, String> attributes) {
+      this.attributes = attributes;
+    }
+
+    public String getMessageId() {
+      return messageId;
```
Comment (anguillanneuf) [inline secretmanager/src/main/java/secretmanager/ConsumeEventNotification.java:69]: the getter/setter boilerplate can be condensed using a `@Data` annotation from lombok: https://projectlombok.org/features/Data

**Review by anguillanneuf — COMMENTED** [review]
java 17 test is flaky, please fix.

File: secretmanager/src/main/java/secretmanager/ConsumeEventNotification.java (hunk 2 above)
Comment (Sita04) [reply secretmanager/src/main/java/secretmanager/ConsumeEventNotification.java]: Done.

File: secretmanager/src/main/java/secretmanager/ConsumeEventNotification.java (hunk 1 above)
Comment (Sita04) [reply secretmanager/src/main/java/secretmanager/ConsumeEventNotification.java:52]: Added it. Thanks.

**Review by anguillanneuf — APPROVED** [review]

**Pull request merged**